.orchestra/config/components/aws-sdk.yml

```diff
@@ -0,0 +1,78 @@
+#@ load("@ytt:overlay", "overlay")
+#@ load("/lib/ninja.lib.yml", "ninja")
+#@ load("/lib/create-component.lib.yml", "single_build_component")
+
+#@yaml/text-templated-strings
+---
+#@ def aws_sdk_args():
+#! version 1.11.113
+#@ target_sha1 = "ece7e32dbca87b3fbc7edf0e81d00f328f4b2b07"
+#@ source_url = "https://github.com/aws/aws-sdk-cpp"
+license: source/LICENSE
+configure: |
+  mkdir -p "$BUILD_DIR"
+  cd "$BUILD_DIR"
+
+  if [[ -f .last-hash && "$(cat .last-hash)" != '(@= target_sha1 @)' ]]; then
+    # Hash changed
+    rm -rf -- source
+  fi
+
+  if [[ ! -f .last-hash && -d source ]]; then
+    # Dirty checkout
+    rm -rf -- source
+  fi
+
+  if [ ! -d source ]; then
```

Comment:
You're trying to be overly smart here, just re-extract the code every time.
That's what we do in all components. If we want to have the feature you're proposing it should be a `revng-orchestra` feature, somehow.

Reply:
I might have gotten a bit carries away :sweat_smile: 
The main problem is that a full checkout, even with `--depth=1` is `1.3G`, which becomes annoying to deal with with consecutive `orc configure` or `orc install`s. In the comment below there's another solution that would solve this and that.

Reply:
Dropped code, implemented via archive downloads